### PR TITLE
[152] Add tests for color-changing menus

### PR DESCRIPTION
**Pull request by GlennPlou**

### CODE REVIEW

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/AbstractRefreshWithCustomizedStyleTest.java [hunk 1]
```diff
@@ -415,14 +416,15 @@ protected void doTestStyleCustomizationThroughColorSelectionFromAppearanceSectio
 
             // Step 2: Enable button and check result
             buttonFromAppearanceSectionToTest.click();
-            assertTrue(BAD_CURRENT_COLOR_BUTTON, bot.button(defaultButtonInColorPalette[i]).isActive());
```
Comment (scosta-obeo) [inline plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/AbstractRefreshWithCustomizedStyleTest.java:418]: You can also remove BAD_CURRENT_COLOR_BUTTON and defaultButtonInColorPalette if you remove this line.

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/sequence/HeaderSequenceDiagramTests.java [hunk 2]
```diff
@@ -85,7 +87,7 @@ protected void onSetUpAfterOpeningDesignerPerspective() throws Exception {
     /**
      * Test if header is correctly renamed after the lifeLine renaming.
      */
-    public void testRenameLifeLine() {
+    public void _testRenameLifeLine() {
```
Comment (scosta-obeo) [inline plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/sequence/HeaderSequenceDiagramTests.java:90]: Why have you deactivated certain tests in this file? They work for me (except for one, but it doesn't seem to have anything to do with colors). Did you forget to reactivate it?

**Review by scosta-obeo — CHANGES_REQUESTED** [review]

File: plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/colors/ColorsAnnotationHelper.java
[the remark this answers is not in the document]
Comment (GlennPlou) [reply plugins/org.eclipse.sirius.diagram.ui/src-diag/org/eclipse/sirius/diagram/ui/tools/internal/colors/ColorsAnnotationHelper.java:1]: OK I applied ifPresent where I could, there are other places where I could use flatMap, but I found that it was affecting the code's readability.

File: plugins/org.eclipse.sirius.tests.swtbot.support/src/org/eclipse/sirius/tests/swtbot/support/api/AbstractSiriusSwtBotGefTestCase.java
[the remark this answers is not in the document]
Comment (GlennPlou) [reply plugins/org.eclipse.sirius.tests.swtbot.support/src/org/eclipse/sirius/tests/swtbot/support/api/AbstractSiriusSwtBotGefTestCase.java:1]: OK I move these new methods in SWTBotSiriusHelper

File: plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/sequence/HeaderSequenceDiagramTests.java (hunk 2 above)
Comment (GlennPlou) [reply plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/sequence/HeaderSequenceDiagramTests.java]: I forgot to reactive ;)

File: plugins/org.eclipse.sirius.tests.swtbot.support/src/org/eclipse/sirius/tests/swtbot/support/api/AbstractSiriusSwtBotGefTestCase.java
[the remark this answers is not in the document]
Comment (scosta-obeo) [reply plugins/org.eclipse.sirius.tests.swtbot.support/src/org/eclipse/sirius/tests/swtbot/support/api/AbstractSiriusSwtBotGefTestCase.java:1]: Copyright 2024 in SWTBotSiriusHelper

**Review by scosta-obeo — APPROVED** [review]

**Pull request merged**